src/main/java/com/google/firebase/remoteconfig/RemoteConfigUtil.java

```diff
@@ -30,10 +31,52 @@ static boolean isValidVersionNumber(String versionNumber) {
     return !Strings.isNullOrEmpty(versionNumber) && versionNumber.matches("^\\d+$");
   }
 
+  static long convertToMilliseconds(String dateString) throws ParseException {
+    if (isUTCDateString(dateString)) {
```

Comment:
This always ends up parsing at least twice. Can we improve that?

```java
try {
  return convertFromUtcZuluFormat(dateString); // common case first
} catch (ParseException e) {
  return convertFromUtcDateFormat(dateString); // fall back
}
```

Reply:
Good point! Thanks!